### PR TITLE
web3.near.transferNear

### PR DESCRIPTION
**Pull request by emaG3m**
**Related Issue(s):** https://github.com/nearprotocol/near-web3-provider/issues/22

**Closes Issue(s):** https://github.com/nearprotocol/near-web3-provider/issues/22

**Nearcore Branch Info:  commit df431768f95bc4117dc7c4d954e38f1b96fd3b9f (HEAD -> stable)

**Unit tests completed?:** Y

**Relevant README/Documentation Updated?:**  Y

### CODE REVIEW

File: src/near_web3_extensions.js [hunk 1]
```diff
@@ -27,13 +27,21 @@ module.exports = function (web3) {
...
-	  property: 'near',
-	  methods: [{
-		  name: 'retrieveNear',
-		  call: 'near_retrieveNear',
-		  params: 1,
-		  inputFormatter: [inputRetrieveNearFormatter],
-		}]
+		property: 'near',
+		methods: [
+			{
```
Comment (emaG3m) [inline src/near_web3_extensions.js:32]: rawr, this extra indentation isn't what's happening in my editor

**Comment by emaG3m** [comment]
@prestwich you may want to make sure you like the naming here:
`web3.near.transferNear`: transfer from evm addr to evm addr of corresponding near accountId
`web3.near.retrieveNear`: transfer near out of evm to near account

**Comment by prestwich** [comment]
Yes, these are good

**Comment by emaG3m** [comment]
> These look good to me. My environment is in a weird state and tests are hanging, but if tests are passing for you then I’m good to merge as soon as merge conflicts are resolved.

hmm....really odd. I wonder if maybe someone else can try and run them and see what happens....

**Comment by prestwich** [comment]
Tests produce this output
```
    console.log
      -----------------------
      Running tests on local network
      NODE_URL: http://localhost:3030
      Account Id: test.near
      Public Key: ed25519:22skMptHjFWNyuEWY22ftn2AbLPSYpmYwGJRGwpNHbTV
      -----------------------

      at Object.<anonymous> (test/near_provider.test.js:31:9)

   console.log
      Account "evm" does not exist

      at accountExists (test/near_provider.test.js:95:21)

    console.log
      Deploying contract on NEAR_ENV: "local"

      at deployContract (test/near_provider.test.js:57:13)

    console.log
      deployed EVM contract Account {
        connection: Connection {
          networkId: 'test',
          provider: JsonRpcProvider { connection: [Object] },
          signer: InMemorySigner { keyStore: [InMemoryKeyStore] }
        },
        accountId: 'evm'
      }

      at deployContract (test/near_provider.test.js:73:17)

    console.log
      { zombieAddress: '0xFc481F4037887e10708552c0D7563Ec6858640d6' }

      at test/near_provider.test.js:215:25

    console.log
      [evm]: evm log: 0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000209e105f40198c000000000000000000000000000000000000000000000000000000000000000667656f7267650000000000000000000000000000000000000000000000000000

      at Account.printLogs (node_modules/near-api-js/lib/account.js:58:21)

    console.log
      [evm]: evm log: 0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000209e105f40198c000000000000000000000000000000000000000000000000000000000000000667656f7267650000000000000000000000000000000000000000000000000000

      at Account.printLogs (node_modules/near-api-js/lib/account.js:58:21)
          at runMicrotasks (<anonymous>)

--------------------------|---------|----------|---------|---------|------------------------------------------------
File                      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                              
--------------------------|---------|----------|---------|---------|------------------------------------------------
All files                 |    80.7 |    66.46 |   78.31 |   80.83 |                                                
 src                      |   80.62 |    66.46 |   78.31 |   80.75 |                                                
  hydrate.js              |    89.8 |       75 |     100 |    89.8 | 35,77,109,131,143                              
  index.js                |   74.48 |    62.65 |      70 |   74.37 | ...450,471,492,513,534,564,623,666,737,805,840 
  near_to_eth_objects.js  |   93.59 |    85.71 |   78.57 |   94.59 | 16,61,156,194                                  
  near_web3_extensions.js |     100 |      100 |     100 |     100 |                                                
  utils.js                |   77.88 |    52.94 |   84.21 |   78.43 | 96-128,171,189,237,291                         
 test                     |     100 |      100 |     100 |     100 |                                                
  config.js               |     100 |      100 |     100 |     100 |                                                
--------------------------|---------|----------|---------|---------|------------------------------------------------

Test Suites: 2 passed, 2 total
Tests:       73 passed, 73 total
Snapshots:   0 total
Time:        29.308s
Ran all test suites matching /test/i.

```

The node produces these errors in addition to normal logging output

```
[/home/james/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmer-runtime-core-0.17.0/src/instance.rs:609] &error = User(
    Any,
)
Jul 30 13:14:48.519  INFO stats: #     574 BRtL8Q9Ht3HmApsBbfEnP9RUNBtTme3PWeLfvhE2fzhs V/1  0/0/40 peers ⬇ 0 B/s ⬆ 0 B/s 4.70 bps 9.91 Tgas/s CPU: 6%, Mem: 133.9 MiB    
[/home/james/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmer-runtime-core-0.17.0/src/instance.rs:609] &error = User(
    Any,
)
Jul 30 13:14:58.521  IN
```

**Comment by barbaraliau** [comment]
Merging, as methods function as expected.

**Pull request merged**